powerauth-java-crypto/src/test/java/io/getlime/security/powerauth/crypto/lib/util/GenerateVectorDataTest.java

```diff
@@ -773,6 +777,95 @@ public void testSignatureValidationV31() throws Exception {
         writeTestVector(testSet);
     }
 
+    @Test
+    public void testSignatureOfflineValidation() throws Exception {
+
+        TestSet testSet = new TestSet("signatures-offline.json", "Client must be able to compute PowerAuth offline signature (using 1FA, 2FA signature keys) based on given data, counter and signature type");
+
+        int min = 2;
+        int max = 8;
+        int keyMax = 2;
+        int signatureCount = 10;
+        int dataMax = 256;
+        for (int j = min; j < max; j++) {
```

Comment:
The configuration and loop above has several flaws:
- Doesn't test for default value equal to 8 (see your test vectors)
- Min value for "j" should start at 4. The actual implementation reject minimal length equal to 4, so we should generate test data from a valid test input.

Reply:
How does the max value of 8 (exclusive) relate to:

```
int signatureDecimalLength = (length == null) ? PowerAuthConfiguration.SIGNATURE_DECIMAL_LENGTH : Math.min(Math.max(length, 4), 12);
```

Reply:
@romanstrobl It does not. Why should it?
@hvge This is not a test, we generate test vectors here.